test/ring/middleware/test/refresh.clj

```diff
@@ -1,6 +1,88 @@
 (ns ring.middleware.test.refresh
-  (:use clojure.test
-        ring.middleware.refresh))
+  (:require
+   [clojure.java.io :as io]
+   [clojure.test :refer [are deftest testing]]
+   [ring.middleware.refresh :as sut]))
 
-(deftest test-wrap-refresh
-  (is (true? false)))
\ No newline at end of file
+(defn create-success-html-handler [body]
+  (constantly {:body body, :headers {"Content-Type" "text/html"}, :status 200}))
+
+(defn invoke-with-get-request [handler]
+  (handler {:request-method :get, :uri "/"}))
+
+(def refresh-script
+  (slurp (io/resource "ring/js/refresh.js")))
+
+(deftest wrap-refresh-adds-script-after-<head>-test
+  (testing "Without a <header> tag"
+   (are [result-body body]
+        (= result-body (-> (create-success-html-handler body)
+                           sut/wrap-refresh
+                           invoke-with-get-request
+                           :body))
+        "<html><body>body</body></html>"
+        "<html><body>body</body></html>"
+
+        "<html><body>body head</body></html>"
+        "<html><body>body head</body></html>"
+
+        "<html><head/><body>body</body></html>"
+        "<html><head/><body>body</body></html>"
+
+        "<html><head /><body>body</body></html>"
+        "<html><head /><body>body</body></html>"
+
+        (str "<html><head>"
+             "<script type=\"text/javascript\">" refresh-script "</script>"
+             "</head><body>body</body></html>")
+        (str "<html><head>"
+             "</head><body>body</body></html>")
+
+        (str "<html><head>"
+             "<script type=\"text/javascript\">" refresh-script "</script>"
+             "<title>title</title></head><body>body</body></html>")
+        (str "<html><head>"
+             "<title>title</title></head><body>body</body></html>")
+
+        (str
```

Comment:
Should follow style guidelines, so:

```clojure
(str x
     y
     z)
```

Or:

```clojure
(str
 x
 y
 z)
```

Reply:
Great catch. Done.